src/app/classify.rs

```diff
@@ -124,6 +128,10 @@ impl classify::ClassifyResponse for Response {
     type ClassifyEos = Eos;
 
     fn start<B>(self, rsp: &http::Response<B>) -> Eos {
+        if rsp.extensions().get::<timeout::ProxyTimedOut>().is_some() {
+            return Eos::Error("timeout");
+        }
```

Comment:
This seems fine, though you might also consider doing this slightly differently: Instead of making the classify module enumerate all of the types of errors it may observfe, we could instead have modules add a `classify::ProxyError(&'static str)` marker onto responses -- so this would be able to work with arbitrary error types in the stack, not just timeouts.

This then becomes something to the effect of:
```rust
        if let Some(ProxyError(msg)) = rsp.extensions().get::<ProxyError>() {
            return Eos::Error(msg);
        }
````

Reply:
That'd seem to have `app` types present in `proxy` modules... Unless you mean just a general `proxy::http::classify::ProxyError` thingy?

Reply:
ah, yeah... good point.